### PR TITLE
New term UI.IsCopyAction

### PR DESCRIPTION
**Pull request by ralfhandl**

### CODE REVIEW

**Review by BerSie — DISMISSED** [review]

**Comment by RaizoC** [comment]
Hmm, now reading it the copy-semantics seems very special and is taken out of the create semantics. I do not understand how this different and more special semantics impacts anything on client side different than the pure "create"-semantics (which I think is very valid). If there really is a use case to handle that one differently (does it need to be unique?) further constraints need to be added to the "copy" semantics, i.e. that it is a bound "create"-action that has the same (bound) entity as result entity.

**Comment by ralfhandl** [comment]
@RaizoC currently we only have a use case for "copy". Should I remove "create" for now, or comment it out?

**Comment by BerSie** [comment]
> Hmm, now reading it the copy-semantics seems very special and is taken out of the create semantics. I do not understand how this different and more special semantics impacts anything on client side different than the pure "create"-semantics (which I think is very valid). If there really is a use case to handle that one differently (does it need to be unique?) further constraints need to be added to the "copy" semantics, i.e. that it is a bound "create"-action that has the same (bound) entity as result entity.

I think the UI can check if an entry was selected before activating the copy function (it knows that an instance has to be selected). For create (simple create without reference) you would not expect that.

**Comment by RaizoC** [comment]
My feedback rather goes into the direction that this is too specific as the client might only be interested in the "create"-semantics. The UI selection depends on whether the action is instance-bound or not, so for this no semantics is needed. What else would the client do with this information?

**Comment by RaizoC** [comment]
I still wonder if these semantics are disjoint, i.e. from the description every "copy" action would also be tagged as "create" action. Is this the intention or does "copy" implies the "create" fact?

**Comment by ralfhandl** [comment]
We can remove "create" for now because https://jira.tools.sap/browse/FIORITECHP1-20377 only requires "copy"

**Review by ArunBharathiKumarasamy — DISMISSED** [review]

**Review by RaizoC — APPROVED** [review]

**Comment by BerSie** [comment]
Why did we put it on the DataFieldForAction and not on the action itself (it should be bound to the action and not its usage/reference)? Sometimes these pull requests are running too long :-) (at least for my old brain).

**Comment by ralfhandl** [comment]
> Why did we put it on the DataFieldForAction and not on the action itself (it should be bound to the action and not its usage/reference)?

Good point!

**Review by RaizoC — DISMISSED** [review]

**Review by RaizoC — APPROVED** [review]
FE already implemented the same.

File: vocabularies/UI.xml [hunk 1]
```diff
@@ -1412,6 +1412,18 @@ The trend is
         </Annotation>
       </Term>
 
+      <Term Name="IsCopyAction" Type="Core.Tag" Nullable="false" DefaultValue="true" AppliesTo="Record">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.RequiresType" String="UI.DataFieldForAction" />
+        <Annotation Term="Core.Description" String="The annotated `DataFieldForAction` record within a `UI.LineItem` annotation references an action that deep-copies an instance of the annotated entity type" />
```
Comment (HeikoTheissen) [inline vocabularies/UI.xml:1418]:
```suggestion
        <Annotation Term="Core.Description" String="The annotated [`DataFieldForAction`](#DataFieldForAction) record within a [`UI.LineItem`](#LineItem) annotation references an action that deep-copies an instance of the annotated entity type" />
```

File: vocabularies/UI.xml [hunk 2]
```diff
@@ -1412,6 +1412,18 @@ The trend is
...
+        <Annotation Term="Core.LongDescription">
+          <String>The referenced action MUST be bound to the annotated entity type and creates a new instance of the same entity type as a deep copy of the bound instance.
```
Comment (HeikoTheissen) [inline vocabularies/UI.xml:1420]:
```suggestion
          <String>The referenced action MUST be bound to the annotated entity type and create a new instance of the same entity type as a deep copy of the bound instance.
```

**Comment by ralfhandl** [comment]
> FE already implemented the same.

I've kept it "experimental", please notify me once this has also been implemented in RAP and proven to work end-to-end.

Thanks in advance!

**Review by HeikoTheissen — CHANGES_REQUESTED** [review]
Add links

**Pull request merged**